### ISSUE TITLE
Access to num_steps and num_epochs to configure the learning rate schedulers

### ISSUE THREAD

**Issue by celsofranssa** [PyTorchLightning/pytorch-lightning#5463]
Some learning rate schedulers like `OneCycleLR`, `CyclicLR` and `CosineAnnealingLR`([torch schedulers](https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate)) requires the number of steps per epoch.

Then, would be great if the number of steps and the number of epochs be placed in the `configure_optimizers(self)` scope.

```
    def configure_optimizers(self, num_steps, num_epochs):

        optimizers = [... ]
        schedulers = [...]

        return optimizers, schedulers
```

**Comment by rohitgr7** [comment]
this might be tricky with `reload_dataloaders_every_epoch` since there is a possibility that `num_steps` might change at each epoch.

**Comment by ayansengupta17** [comment]
For `Lightning 2.0`
```
def num_steps(self) -> int:
        
        dataset = self.trainer.fit_loop._data_source.dataloader()
        dataset_size = len(dataset)
        num_devices = max(1, self.trainer.num_devices)
        num_steps = dataset_size * self.trainer.max_epochs // (self.trainer.accumulate_grad_batches * num_devices)
        return num_steps

```
